cs/src/Contracts/TunnelPort.cs

```diff
@@ -4,109 +4,124 @@
-        /// to the individual port.
-        /// </remarks>
-        /// <seealso cref="TunnelAccessScopes" />
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public IDictionary<string, string>? AccessTokens { get; set; }
-
-        /// <summary>
-        /// Gets or sets access control settings for the tunnel port.
-        /// </summary>
-        /// <remarks>
-        /// See <see cref="TunnelAccessControl" /> documentation for details about the
-        /// access control model.
-        /// </remarks>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public TunnelAccessControl? AccessControl { get; set; }
-
-        /// <summary>
-        /// Gets or sets options for the tunnel port.
-        /// </summary>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public TunnelOptions? Options { get; set; }
-
-        /// <summary>
-        /// Gets or sets current connection status of the tunnel port.
-        /// </summary>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public TunnelPortStatus? Status { get; set; }
-
-        /// <summary>
-        /// Gets or sets the username for the ssh service user is trying to forward.
-        /// </summary>
-        /// <remarks>
-        /// Should be provided if the <see cref="TunnelProtocol"/> is Ssh.
-        /// </remarks>
-        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
-        public string? SshUser { get; set; }
     }
+
+    /// <summary>
+    /// Gets or sets the ID of the cluster the tunnel was created in.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RegularExpression(ClusterIdPattern)]
+    [StringLength(ClusterIdMaxLength, MinimumLength = ClusterIdMinLength)]
+    public string? ClusterId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the generated ID of the tunnel, unique within the cluster.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RegularExpression(TunnelIdPattern)]
+    [StringLength(TunnelIdLength, MinimumLength = TunnelIdLength)]
+    public string? TunnelId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the IP port number of the tunnel port.
+    /// </summary>
+    public ushort PortNumber { get; set; }
+
+    /// <summary>
+    /// Gets or sets the optional short name of the port.
+    /// </summary>
+    /// <remarks>
+    /// The name must be unique among named ports of the same tunnel.
+    /// </remarks>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [RegularExpression(TunnelNamePattern)]
+    [StringLength(TunnelNameMaxLength)]
+    public string? Name { get; set; }
+
+    /// <summary>
+    /// Gets or sets the optional description of the port.
+    /// </summary>
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    [StringLength(DescriptionMaxLength)]
```

Comment:
Do we have disallowed characters for tunnel descriptions? Perhaps we shouldn't allow html-control characters <, &, >, and maybe also ", ', /, and \\

Reply:
Depending on the context there are many characters that can be dangerous, and I think it's unreasonable to block or strip all of them in a description. Since we cannot strictly validate this property, the proper mitigation is to ensure that anywhere the description is rendered that it is properly encoded/escaped at that point.